design/20220118.certificate-issuance-exponential-backoff.md

```diff
@@ -0,0 +1,144 @@
+# Exponential backoff for certificate issuance
+
+## Summary
+
+Implement a way to apply exponential backoff when retrying failed certificate
+issuances. Issuance in this context refers to the period of time during which
+the `Issuing` condition on a `Certificate` is set to true and for which a new
+set of issuance-specific resources (`CertificateRequest`s, `Order`s, `Challenge`s etc) is
+created.
+
+## Motivation
+
+Currently failed issuances are retried once an hour without a backoff or time limit. This means that 1) continuous failures in large installations can overwhelm external services 2) in case of some external problems (i.e DNS) that cause issuances to fail and last for longer than a couple of hours, rate limits (see [Let'sEncrypt rate limts](https://letsencrypt.org/docs/rate-limits/)) can be easily hit.
+
+## Goals
+
+- Ensure that retrying failed issuances does not overwhelm external services and is less likely to cause issuance being rate-limited by backing off with an exponentially increasing time period. 
+
+- Ensure that when the backoff is being applied, users have a way to find out when the issuance will be next retried and that the backoff mechanism does not introduce extra complexity for debugging and fits in with the already existing 1 hour backoffs
+
+- Ensure that cmctl renew can still be used to manually force an immediate issuance attempt, so that in cases where an issuance was failing due to a setup error (i.e DNS setup) and a user believes that they have fixed it, they have a way to verify the fix without waiting
+
+## Non-goals
+
+- Make the backoff period configurable (extra complexity). For context Kubernetes pod backoff crashloopbackoff period is _not_ configurable (although it is a very demanded feature [k/k#57291](https://github.com/kubernetes/kubernetes/issues/57291))
+
+- Make it possible to reset the backoff period _without_ having a successful issuance first (i.e by manually running `cmctl renew`)
```

Comment:
To be clear, it *is* possible to trigger an immediate *attempt*, but at the end of that attempt the counter *won't* be reset to zero (unless it succeeded), therefore the next issuance after that will have a greater back-off applied, right?